src/mds/Server.cc

```diff
@@ -1593,6 +1606,9 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
     mds->logger->inc(l_mds_reply);
     utime_t lat = ceph_clock_now() - mdr->client_request->get_recv_stamp();
     mds->logger->tinc(l_mds_reply_latency, lat);
+    if (client_inst.name.is_client()) {
+      mds->sessionmap.hit_session(session);
```

Comment:
session can be null

Reply:
even if the request is from a client? In `Server::handle_client_request()`, `mdr->session` is initialized if its a client request. The request is dropped if session is not valid (i.e., nullptr). So, I think its safe to access `mdr->session` here.